Drivers/SensorFusion/Src/SensorFusion.cpp

```diff
@@ -534,19 +537,22 @@ SFError_t SF_GenerateNewResult()
     #endif
 
 #endif
+    xSemaphoreGive(SF_mutex);
     return SFError;
 }
 
-
-
-SFError_t SF_GetResult(SFOutput_t *output)
+SFError_t SF_GetResult(SFOutput_t* output)
 {
     SFError_t SFError;
 
     //TODO: This function always return success as return, fix it?
     SFError.errorCode = 0;
 
-    *output = SFOutput;
+    xSemaphoreTake(SF_mutex, portMAX_DELAY);
+
+    output = &SFOutput;
```

Comment:
A mutex wouldn't be required for this operation, since you're providing a reference to the output. We really should be doing a copy here instead so that the caller will get it's own copy of the data that it can safely interact with.